tensorflow/python/ops/script_ops.py

```diff
@@ -702,12 +700,21 @@ def numpy_function(func, inp, Tout, name=None):
     inp: A list of `tf.Tensor` objects.
     Tout: A list or tuple of tensorflow data types or a single tensorflow data
       type if there is only one, indicating what `func` returns.
+    stateful: (Boolean.) If True, the function should be considered stateful and
+      may has side effects. If a function is stateless, when given the same
+      input it will return the same output and have no observable side effects.
+      The behavior for a stateful function with the `stateful` argument False is
+      undefined; in particular, caution should be taken not
+      to mutate the input arguments as this is a stateful operation.
+      Optimizations such as common subexpression elimination are only performed
```

Comment:
Let's make a strnger distinction between "stateless operation" = thing your code might do that mutates some state and "stateful=False" = the user set this flac to False.

Specifically, the last sentence should say "Optimizations [...] are only performed when the `stateful` flag is set to `False`"

Reply:
Okay sure, I'll gladly change this. But the docs is taken one-by-one from `py_func_common`. Should this here be more "end user facing"? Or why would we not want to have the same docstring?

Reply:
Definitely. Unlike py_func_common, this docstring will appear on the tensorflow.org website in the public documentation.

Reply:
I see, sure. I rewrote the docs completely to make it clearer, let me know what you think